Make flashes less effective

babydoll opened the pull request:
:cl: sabiram
balance: Flashes have been reworked when attacking humanoid targets.
balance: Instead of stunning for 20 seconds, they will stun for up to 7 seconds, and blur vision and confuse for up to 18 seconds.
balance: Flash stun duration for silicons has been reduced from 10-20 seconds to 8-12 seconds.
tweak: The area-of-effect attack for flashes now blurs the eyes of those affected, in addition to flashing them.
rscadd: Flashes can now have varying effects by type.
tweak: The flash used in robot construction is now weaker than the base flash, but breaks just as often, instead of every time its used.
tweak: The spray projectile fired from peppersprays now moves much faster and their capacity has been increased from 4 shots to 6.
tweak: Pepper spray now incapacitates for an additional 2 seconds.
/:cl:
![](https://i.imgur.com/BVHhdaS.gif)

babydoll committed:
Make flashes less effective

ghost commented:
Does not remove the fundamental problem behind flashes but is a step in the right direction.

Techhead0 commented:
The other thing to consider is changing `Weaken()` to `Stun()` so that you still can't move or act, but at least you don't fall over and drop all your shit. Can be combined with other changes, like just-blind/blind-and-stunned/blind-and-weakened on different levels of flash strength.

EDIT: Anyway, I'm in favor, but leaving this up for other people to weigh in before I review it.

Crushtoe commented:
Flashbangs still work fine?

ghost commented:
If different flash-levels are implemented consider adding a longer recharge delay so people can't spam the flash until someone falls over.

hey-it-stev commented:
Why are we both nerfing sunglasses and what they counter? That's, like, cancelling each other out balance-wise. If flashes are nerfed significantly remove the sunglasses overlay.

ghost commented:
Ah, the flash. Humble tool of security, coveted by those of less savory persuasions. Perhaps this will force people to seek out other means of instantly stunning people, perhaps not. But at least it'll become less of a enabler of quick and easy murders if this gets merged.

babydoll commented:
This is a work in progress

Techhead0 commented on code/game/objects/items/devices/flash.dm:
```diff
@@ -69,18 +69,25 @@
+				var/flash_strength = (rand(2,7))
 				if(ishuman(M))
 					var/mob/living/carbon/human/H = M
 					flash_strength *= H.species.flash_mod
```
This didn't get it before, but it should probably get a `round()`

Techhead0 replied:
May want to use either `round()` or `ceil()` here now that non-integers can happen.
(All `flash_mod` values are currently multiples of 0.1, so it wasn't a worry when multiplying by 10.)

Techhead0 commented on code/game/objects/items/devices/flash.dm:
```diff
@@ -69,18 +69,25 @@
-				if(flash_strength > 0)
-					M.Weaken(flash_strength)
+				if(flash_strength > 0) // always flash, stun, blur, and confuse
+					world << "Flashed with [flash_strength]"
```
Make sure you remove this when you do your final version.

Techhead0 commented on code/game/objects/items/devices/flash.dm:
```diff
@@ -69,18 +69,25 @@
+					M.eye_blurry += flash_strength
+					M.confused += flash_strength // 4 to 12 seconds of confusion and blurred vision
+					if(flash_strength > 3) // often cause them to drop items - 4, 5
+						M.drop_item()
```
`drop_item()` only drops what's in their active hand. It's usually used in `attackby()` procs.

Techhead0 commented on code/game/objects/items/devices/flash.dm:
```diff
@@ -69,18 +69,25 @@
+					if(flash_strength > 3) // often cause them to drop items - 4, 5
+						M.drop_item()
+					if(flash_strength > 5) // only fully floor them rarely - 6 & 7
+						M.Weaken(flash_strength) // 12 to 14 seconds of weaken
```
Not sure about this, as it makes a big difference between the effects of 5 and 6.

babydoll commented:
Might be worth investigating porting these. I tried just now but I couldn't get it working.
https://github.com/PolarisSS13/Polaris/pull/2134
https://github.com/PolarisSS13/Polaris/pull/2897

PurpleMartinJCK reviewed (commented):
If flashes are becoming weaker then the negative effects on those who are weak to flashes also needs to be adjusted. I'm not blocking this as a developer but as a species maintainer so I'm not using request changes but this should be addressed before it's merged. https://github.com/Baystation12/Baystation12/blob/ebf4521bd6bae352122cc54d55bc66202f94ebe2/code/modules/organs/subtypes/nabber.dm#L24

PurpleMartinJCK commented:
This is also a pretty major change so please leave 24 hours for more comments before this is merged as I could see it inciting discussion.

Actually just had all the thumbs downs and stuff pointed out to me (tired, sorry). So I'm gonna mark this controversial. Could you please open a new thread in our new gameplay subforum? Then link it back here. All future discussion about the benefits or drawbacks of the PR should go there once posted.

Rebruiza commented:
No. Sunglasses exist for a reason, and flashes are good for a "stop this right now" type of weapon, instead of relying on a heart damage caused by pain from a taser.

babydoll committed:
Adjust pepperspray projectile speed

Also further reduces stun duration for silicons and changes
a number for GAS.

comma commented:
why do we have two laels that mean same ree

ghost commented:
because xales loves their special snowflake label that's pink :^)

babydoll committed:
Further changes to flashes and pepper spray

Pepper spray's projectile now moves as quickly as permitted
Flashes can now have stronger or weaker effects by subtype
The synthetic flash used in robot construction now has a weaker effect
instead of breaking every time
Added an 'advanced flash' as concept - presently unavailable in game

comma commented on code/modules/reagents/reagent_containers/spray.dm:
```diff
@@ -15,6 +15,7 @@
 	possible_transfer_amounts = "5;10" //Set to null instead of list, if there is only one.
 	var/spray_size = 3
 	var/list/spray_sizes = list(1,3)
+	var/projspeed = 10 // lower is faster
```
rename to step_delay IMO, pls no more vars named opposite way

PurpleMartinJCK reviewed (changes requested):
Contention resolved, the PR can be merged once it does the following:
* Increase the stun time on pepper spray ~~and stun baton is increased~~ slightly (1-2 seconds)
* Make pepper spray moved faster to avoid self spraying
* Reduce species specific impacts of flashes at request of the species maintainer

Edit: stun baton changes are too far reaching for the scope of this PR

comma commented:
>Make pepper spray moved faster to avoid self spraying

Already done, it's maybe even /too/ fast.

>Reduce species specific impacts of flashes at request of the species maintainer

Can be done when such requests actually, you know, happen. Most of it is factored in with flash_mod var of species, only GAS had snowflake handling for it.

PurpleMartinJCK commented:
Some of the changes were done while the discussion was ongoing, this is just to show all the things that were changed as a result of that discussion

babydoll committed:
Make additional changes to pepper spray

Stun duration from pepper spray increased by 2 seconds
Projectile speed var renamed for clarity
Fixed introduced bug - flash icon states not being named correctly
